tools/ci/src/commands/compile_check_no_std.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{Flag, Prepare, PreparedCommand};
+use argh::FromArgs;
+use xshell::cmd;
+
+/// Checks that the project compiles for a `no_std` target.
+#[derive(FromArgs, Default)]
+#[argh(subcommand, name = "compile-check-no-std")]
+pub struct CompileCheckNoStdCommand {}
+
+impl Prepare for CompileCheckNoStdCommand {
+    fn prepare<'a>(&self, sh: &'a xshell::Shell, _flags: Flag) -> Vec<PreparedCommand<'a>> {
+        vec![PreparedCommand::new::<Self>(
```

Comment:
Can we reduce the amount of copy-pasting that will happen as more crates are added here by building the vec of commands from a vec of crate name strings?

Reply:
We _could_, but some crates will require features as well (e.g., `bevy_mikktspace` requires the `libm` feature). Once we have `no_std` exposed through `bevy` itself rather than its crates we'll be able to remove all these individual checks anyway, since it would be sufficient to just check that `bevy` itself works on `no_std`